torch/csrc/distributed/rpc/init.cpp

```diff
@@ -684,6 +684,12 @@ PyObject* rpc_init(PyObject* _unused, PyObject* noargs) {
           (std::vector<WorkerInfo>(TensorPipeAgent::*)() const) &
               TensorPipeAgent::getWorkerInfos,
           py::call_guard<py::gil_scoped_release>())
+      .def(
+          "_get_device_map",
```

Comment:
curious, do you need the device map info, or all you need is `_supports_cuda`?

Reply:
Thanks for pointing this out! I don't have much context here.

Is `_supports_cuda` here equivalent to `device_map` not empty?

@lw Any suggestion?

Reply:
If we want to be precise there are various levels of detail. We could have a single boolean `_supports_cuda` field which says whether the agent understands CUDA tensors. We could have a `_devices` fields which says which devices the agent is allowed to operate on (and, if this list contains a CUDA device, then the agent supports CUDA). Or we could have `_devices_for_remote(remote_worker)` which returns the set of devices that can be used when making an RPC call to a certain worker (which would be the keys of the device map for that worker, and would be a subset of `_devices`). Or we could just expose the full device map.

Any of the option above "works", it just depends what @SciPioneer needs.

Reply:
> If we want to be precise there are various levels of detail. We could have a single boolean `_supports_cuda` field which says whether the agent understands CUDA tensors. We could have a `_devices` fields which says which devices the agent is allowed to operate on (and, if this list contains a CUDA device, then the agent supports CUDA). Or we could have `_devices_for_remote(remote_worker)` which returns the set of devices that can be used when making an RPC call to a certain worker (which would be the keys of the device map for that worker, and would be a subset of `_devices`). Or we could just expose the full device map.
> 
> Any of the option above "works", it just depends what @SciPioneer needs.

Thanks for the detailed explanation! My current implementation exposes the full device map of the destination worker to `RemoteModule`, and it checks whether the device map is empty. I verified that this works for me.